R/coord-cartesian-.r

```diff
@@ -79,50 +79,124 @@ CoordCartesian <- ggproto("CoordCartesian", Coord,
   is_free = function() TRUE,
 
   distance = function(x, y, panel_params) {
-    max_dist <- dist_euclidean(panel_params$x.range, panel_params$y.range)
+    max_dist <- dist_euclidean(panel_params$x.scale$dimension(), panel_params$y.scale$dimension())
     dist_euclidean(x, y) / max_dist
   },
 
   range = function(panel_params) {
-    list(x = panel_params$x.range, y = panel_params$y.range)
+    list(x = panel_params$x.scale$dimension(), y = panel_params$y.scale$dimension())
   },
 
   backtransform_range = function(self, panel_params) {
     self$range(panel_params)
   },
 
   transform = function(data, panel_params) {
-    rescale_x <- function(data) rescale(data, from = panel_params$x.range)
-    rescale_y <- function(data) rescale(data, from = panel_params$y.range)
-
-    data <- transform_position(data, rescale_x, rescale_y)
+    data <- transform_position(data, panel_params$x.scale$map, panel_params$y.scale$map)
     transform_position(data, squish_infinite, squish_infinite)
   },
 
   setup_panel_params = function(self, scale_x, scale_y, params = list()) {
-    train_cartesian <- function(scale, limits, name) {
-      range <- scale_range(scale, limits, self$expand)
+    c(
+      view_scales_from_scale(scale_x, self$limits$x, self$expand),
```

Comment:
I think it would be cleaner to have name this list, and not embedded the aesthetic name in the list returned by `view_scales_from_scale()`, so the code below could be `panel_params$x$scale$break_positions_minor()`